test/GoogleUtils.Tests/Streaming/GPSStreamTests.cs

```diff
@@ -0,0 +1,169 @@
+﻿using Orleans.Providers.Streams;
+using Orleans.Runtime.Configuration;
+using Orleans.Storage;
+using Orleans.TestingHost;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using TestExtensions;
+using UnitTests.Streaming;
+using UnitTests.StreamingTests;
+using Xunit;
+
+namespace GoogleUtils.Tests.Streaming
+{
+    [TestCategory("GCP"), TestCategory("PubSub")]
+    public class GPSStreamTests : TestClusterPerTest
+    {
+        public static readonly string PUBSUB_STREAM_PROVIDER_NAME = "GPSProvider";
+        private SingleStreamTestRunner runner;
+
+        public override TestCluster CreateTestCluster()
+        {
+            if (!GoogleTestUtils.IsPubSubSimulatorAvailable.Value)
+            {
+                throw new SkipException("Google PubSub Simulator not available");
+            }
+
+            var options = new TestClusterOptions();
+            //from the config files
+            options.ClusterConfiguration.AddMemoryStorageProvider("MemoryStore", numStorageGrains: 1);
+            options.ClusterConfiguration.AddSimpleMessageStreamProvider("SMSProvider", fireAndForgetDelivery: false);
+            options.ClientConfiguration.AddSimpleMessageStreamProvider("SMSProvider", fireAndForgetDelivery: false);
+
+            var providerSettings = new Dictionary<string, string>
+                {
+                    { "ProjectId",  GoogleTestUtils.ProjectId },
+                    { "TopicId",  GoogleTestUtils.TopicId },
+                    { "DeploymentId",  GoogleTestUtils.DeploymentId.ToString()},
+                };
+
+            options.ClientConfiguration.RegisterStreamProvider<GooglePubSubStreamProvider>("GPSProvider", providerSettings);
+
+            options.ClusterConfiguration.Globals.RegisterStreamProvider<GooglePubSubStreamProvider>("GPSProvider", providerSettings);
+            options.ClusterConfiguration.Globals.RegisterStreamProvider<GooglePubSubStreamProvider>("GPSProvider2", providerSettings);
```

Comment:
what's the reason to configure extra sms stream provider and second `GPSProvider2 `when only `GPSProvider `is used in the test?